holoviews/operation/datashader.py

```diff
@@ -1565,7 +1565,8 @@ def _process(self, element, key=None):
             new_data[tuple(vd.name for vd in vdims)] = img
         else:
             new_data = array
-        return element.clone(new_data, **kwargs)
+        return element.clone(new_data, xdensity=element.xdensity,
+                             ydensity=element.ydensity, **kwargs)
```

Comment:
Why does `clone` not already copy `xdensity` and `ydensity` from what it is cloning?

Reply:
Would have to be conditional on whether or not the data is passed and it's cheap enough to compute from the bounds and the data shape, but the immediate reason is that they are not parameters.